### ISSUE TITLE
Could Impact Analysis include Public Groups?

### ISSUE THREAD

**Issue by godfroy316** [pgonzaleznetwork/sfdc-happy-soup#139]
Ability to see public group sharing usage related to things like list views, report folders, etc would be awesome!

**Comment by pgonzaleznetwork** [comment]
Hi @godfroy316 ,

Thanks for reaching out! Can you please expand on your use case? List views and report folders are not available in HappySoup at the moment so I'm curious what your scenario is.

Also, to thank you for reaching out, I'm giving you the exclusive link to HappySoup 2.0 BETA https://happysoup2-beta.herokuapp.com/

**Comment by jamienrogers222** [comment]
Hi @pgonzaleznetwork, @godfroy316 is on my team. He's referring to similar functionality of the Where Is It Used button, but on Public Groups. We this idea posted on the ideaexchange - https://trailblazer.salesforce.com/ideaView?id=08730000000BrQjAAK

It would be AMAZING if we could find all the places a Public Group is used using Happy Soup.

**Comment by pgonzaleznetwork** [comment]
@jamienrogers222 Thanks for the added info. This is indeed a great idea, I'll look into it!

**Comment by godfroy316** [comment]
Yes!  Thank you @jamienrogers222 and @pgonzaleznetwork!  Apologies for not following up with this.

**Comment by zerkz** [comment]
From my surface level research, it doesn't look like the Metadata dependency API supports this yet. But you might be able to gleam usages by doing the following: 

- Querying share records for that group ID
- Simple file search across your git repositories holding your metadata.

**Comment by pgonzaleznetwork** [comment]
Thank you @zerkz ! I suspected this wasn't available via the dependency API as support is generally limited. You are right, I would have to query records for that public group, and also the SharingRule object to show specific rules https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_sharingrules.htm
